src/tests/unit-tests.cpp

```diff
@@ -1,6 +1,64 @@
+#include "../app/doctor/doctor.hpp"
+#include "../app/visit/visit.hpp"
 #include "gtest/gtest.h"
 
-TEST(HelloWorld, Hello) {
-    ASSERT_TRUE(true);
+struct AssociationTest : ::testing::Test
```

Comment:
I would consider using class instead of struct as it is private by default and we would like to hide it from "the world"
 
 Also name could be called AssociationTestFixture (it directly says that's the fixture).
 I would prefer fixture to in separate file. 
 Fixture in .hpp file, tests in .cpp file
 It is a good practice to close tests within anonymous namespace 
 [CppCoreGuidlines](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#sf22-use-an-unnamed-anonymous-namespace-for-all-internalnon-exported-entities)

Reply:
There would be more suggestions for the tests, but a this point I'd like to create some training and then we can refactor those.

Reply:
Yes, the test will need to be refactored, but for now gonna leave it as is. I've created some basic tests just to demonstrate to everyone how associations and extensions should function.